dbtwiz/bigquery.py

```diff
@@ -1,121 +1,152 @@
+        """Fetch column names and descriptions from BigQuery."""
+        table_ref = f"{project}.{dataset}.{table_name}"
+        try:
+            table = self.get_client().get_table(table_ref)
+            columns = self.parse_schema(table.schema)
+            return columns, ""
+        except self.NotFound:
+            return None, f"Error: The table '{table_name}' does not exist in BigQuery."
+        except self.Forbidden:
+            return None, f"Error: You do not have access to the table '{table_name}'."
+        except Exception as e:
+            return None, f"Error: Failed to fetch table details from BigQuery: {e}"
+
+    def check_project_exists(self, project) -> str:
+        """Checks whether the given project exists in BigQuery"""
+        try:
+            # Check if the project exists and is accessible
+            datasets = list(self.get_client().list_datasets(project=project))
+            if not datasets:
+                return (
+                    f"Warning: The project '{project}' exists but contains no datasets."
+                )
+            else:
+                return "Exists"
+        except self.NotFound:
+            return f"Error: The project '{project}' does not exist."
+        except self.Forbidden:
+            return f"Error: You do not have access to the project '{project}'."
+        except Exception as e:
+            return f"Error: Failed to verify project '{project}': {e}"
+
+    def run_query(self, project, query):
+        """Runs a query in bigquery"""
+        return self.get_client().query(query, project=project)
+
+    def delete_table(self, table_id, project=None):
+        """Deletes a table from bigquery"""
+        self.get_client().delete_table(table_id, project=project)
+
+    def get_bigquery_partition_expiration(self, table_id: str) -> int:
+        """Get the current partition expiration for a table in BigQuery."""
+        table = self.get_client().get_table(table_id)
+        if (
+            table.time_partitioning
+            and table.time_partitioning.expiration_ms is not None
+        ):
+            return table.time_partitioning.expiration_ms // (
+                1000 * 60 * 60 * 24
+            )  # Convert ms to days
+        return -1  # Return -1 if no expiration is set
+
+    def update_bigquery_partition_expiration(self, table_id: str, expiration_days: int):
+        """Update the partition expiration for a table in BigQuery."""
+        table = self.get_client().get_table(table_id)
+        if table.time_partitioning:
+            # Create a new TimePartitioning object with the updated expiration
+            updated_partitioning = self.get_bigquery().TimePartitioning(
+                type_=table.time_partitioning.type_,
+                field=table.time_partitioning.field,
+                expiration_ms=expiration_days
+                * 24
+                * 60
+                * 60
+                * 1000,  # Convert days to ms
```

Comment:
Consider moving occurences of these to a `MILLISECONDS_PER_DAY` constant for readability.

Reply:
Good suggestion!